mirrord/layer/src/common.rs

```diff
@@ -11,29 +12,77 @@ use crate::{
     HOOK_SENDER,
 };
 
+/// Type alias for a queue of responses from the agent, where these responses are [`RemoteResult`]s.
+///
+/// ## Usage
+///
+/// Most mirrord features that interact with the agent use this type to keep a queue of the
+/// operations, keeping them in order, and allowing us to match a hook request to a response. It's
+/// done in this way due to us not having identifiers for the hook requests.
+///
+/// - The usual flow is:
+///
+/// 1. `push_back` the [`oneshot::Sender`] that will be used to produce the [`RemoteResult`];
+///
+/// 2. When the operation gets a response from the agent:
+///     1. `pop_front` to get the [`oneshot::Sender`], then;
+///     2. `Sender::send` the result back to the operation that initiated the request.
 pub(crate) type ResponseDeque<T> = VecDeque<ResponseChannel<T>>;
 
+/// Sends a [`HookMessage`] through the global [`HOOK_SENDER`].
+///
+/// ## Usage
+///
+/// - [`file::ops`](crate::file::ops): most of the file operations are blocking, and thus this
+///   function is extensively used there;
+///
+/// - [`socket::ops`](crate::socket::ops): used by some functions that are _blocking-ish_.
 pub(crate) fn blocking_send_hook_message(message: HookMessage) -> HookResult<()> {
     HOOK_SENDER
         .get()
         .ok_or(HookError::EmptyHookSender)
         .and_then(|hook_sender| hook_sender.blocking_send(message).map_err(Into::into))
 }
 
+/// Type alias for the channel that sends a response from the agent.
+///
+/// See [`ResponseDeque`] for usage details.
 pub(crate) type ResponseChannel<T> = oneshot::Sender<RemoteResult<T>>;
 
+/// Hook message for the `socket::getaddrinfo` operation.
```

Comment:
Why is this the only internal hook struct defined here?

Reply:
Not sure we have an actual good place for it. Should I move it to the one of the tcp modules we have?

Reply:
Why not give DNS its own module?
Also why is the naming inconsistent here?
`  Tcp(HookMessageTcp),
    TcpOutgoing(TcpOutgoing),
    UdpOutgoing(UdpOutgoing),
    File(HookMessageFile),
    GetAddrInfoHook(GetAddrInfoHook),`

Reply:
- dns module:

Seems like a good idea to me, should I refactor this?

- Naming:

Good question. Should I rename these (suggestions)?

- `Tcp(HookMessageTcp)` -> `Tcp(TcpIncoming)`;
- `GetAddrInfoHook(GetAddrInfoHook)` -> `GetAddrInfo(GetAddrInfo)`;
- `File(HookMessageFile)` -> `File(FileOperation)`;

Or should I turn them all into `HookMessageX`, like the `HookMessageTcp`?

Reply:
DNS module - yes
Naming: first option IMO

Reply:
Done, and done.